cmd/agent/common/commands/check.go

```diff
@@ -0,0 +1,567 @@
+	cmd.Flags().IntVarP(&checkTimes, "check-times", "t", 1, "number of times to run the check")
+	cmd.Flags().IntVar(&checkPause, "pause", 0, "pause between multiple runs of the check, in milliseconds")
+	cmd.Flags().StringVarP(&logLevel, "log-level", "l", "", "set the log level (default 'off') (deprecated, use the env var DD_LOG_LEVEL instead)")
+	cmd.Flags().IntVarP(&checkDelay, "delay", "d", 100, "delay between running the check and grabbing the metrics in milliseconds")
+	cmd.Flags().BoolVarP(&formatJSON, "json", "", false, "format aggregator and check runner output as json")
+	cmd.Flags().StringVarP(&breakPoint, "breakpoint", "b", "", "set a breakpoint at a particular line number (Python checks only)")
+	cmd.Flags().BoolVarP(&profileMemory, "profile-memory", "m", false, "run the memory profiler (Python checks only)")
+	cmd.Flags().BoolVar(&fullSketches, "full-sketches", false, "output sketches with bins information")
+	config.Datadog.BindPFlag("cmd.check.fullsketches", cmd.Flags().Lookup("full-sketches")) //nolint:errcheck
+
+	// Power user flags - mark as hidden
+	createHiddenStringFlag(cmd, &profileMemoryDir, "m-dir", "", "an existing directory in which to store memory profiling data, ignoring clean-up")
+	createHiddenStringFlag(cmd, &profileMemoryFrames, "m-frames", "", "the number of stack frames to consider")
+	createHiddenStringFlag(cmd, &profileMemoryGC, "m-gc", "", "whether or not to run the garbage collector to remove noise")
+	createHiddenStringFlag(cmd, &profileMemoryCombine, "m-combine", "", "whether or not to aggregate over all traceback frames")
+	createHiddenStringFlag(cmd, &profileMemorySort, "m-sort", "", "what to sort by between: lineno | filename | traceback")
+	createHiddenStringFlag(cmd, &profileMemoryLimit, "m-limit", "", "the maximum number of sorted results to show")
+	createHiddenStringFlag(cmd, &profileMemoryDiff, "m-diff", "", "how to order diff results between: absolute | positive")
+	createHiddenStringFlag(cmd, &profileMemoryFilters, "m-filters", "", "comma-separated list of file path glob patterns to filter by")
+	createHiddenStringFlag(cmd, &profileMemoryUnit, "m-unit", "", "the binary unit to represent memory usage (kib, mb, etc.). the default is dynamic")
+	createHiddenStringFlag(cmd, &profileMemoryVerbose, "m-verbose", "", "whether or not to include potentially noisy sources")
+
+	cmd.SetArgs([]string{"checkName"})
+}
+
+// Check returns a cobra command to run checks
+func Check(loggerName config.LoggerName, confFilePath string, flagNoColor bool, clusterAgent bool) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "check <check_name>",
+		Short: "Run the specified check",
+		Long:  `Use this to run a specific check with a specific rate`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			configName := ""
+			if clusterAgent {
+				// we'll search for a config file named `datadog-cluster.yaml`
+				configName = "datadog-cluster"
+			}
```

Comment:
I recently created the [pkg/util/flavor](https://github.com/DataDog/datadog-agent/blob/master/pkg/util/flavor/flavor.go) package, a general `configName` could be set up there when initializing the flavor and retrieved here (that way we could get rid of the `clusterAgent` bool)

Reply:
That's great, good to know thank you!
I guess this does imply doing the same for all agent flavors and touching too many packages to make it consistent, I'm not comfortable doing it in this PR as it's out of scope + I'm not code owner 😅

Reply:
I am not completely familiar with the code you are modifying here but I don't think you need to change all flavors for this: you can set up `ConfigurationName` as the empty string and change it only when the flavor is the cluster agent one in the `SetFlavor` function.

Nonetheless, it is only a suggestion, we can work on unifying flavors on a different PR.

Reply:
sounds good, I'll see what I can do

Reply:
@mx-psi now I rely on the `flavor` pkg and got rid of the `clusterAgent` bool argument as you suggested.
I preferred not to add `ConfigurationName` to the `flavor` pkg though as I believe if we do we have to use it everywhere instead of hardcoded config names. I do agree that your suggestion about `ConfigurationName` is cleaner but as I said I believe it's out of scope for this PR + I'm not owner of these packages.
Let me know wdyt of the current change and please backlog something about for platform or core if you think it's something that we have to do.
Thanks again for the suggestion!

Reply:
Looks good to me! I will discuss removing the hardcoded config names on a separate PR with core :)

Reply:
thank you